ingestion-beam/src/main/java/com/mozilla/telemetry/aet/DecryptAetIdentifiers.java

```diff
@@ -0,0 +1,289 @@
+
+    private void processDesktopTelemetryPayload(ObjectNode json) throws IOException, JoseException {
+      validator.validate(telemetryAetSchema, json);
+      System.out.println("Go telemetry");
+      ObjectNode payload = (ObjectNode) json.path(FieldName.PAYLOAD);
+      JsonNode anonIdNode = payload.remove("ecosystemAnonId");
+      if (anonIdNode != null) {
+        payload.set("ecosystemUserId", decrypt(keyStore, anonIdNode));
+      }
+      ArrayNode prevAnonIdsNode = (ArrayNode) payload.remove("previousEcosystemAnonIds");
+      if (prevAnonIdsNode != null) {
+        payload.set("previousEcosystemUserIds", decrypt(keyStore, prevAnonIdsNode));
+      }
+    }
+
+    private void processStructuredIngestionPayload(ObjectNode json)
+        throws IOException, JoseException {
+      validator.validate(structuredAetSchema, json);
+      JsonNode anonIdNode = json.remove("ecosystem_anon_id");
+      if (anonIdNode != null) {
+        json.set("ecosystem_user_id", decrypt(keyStore, anonIdNode));
+      }
+      ArrayNode prevAnonIdsNode = (ArrayNode) json.remove("previous_ecosystem_anon_ids");
+      if (prevAnonIdsNode != null) {
+        json.set("previous_ecosystem_user_ids", decrypt(keyStore, prevAnonIdsNode));
+      }
+    }
+
+    @Override
+    public Iterable<PubsubMessage> apply(PubsubMessage message) {
+      message = PubsubConstraints.ensureNonNull(message);
+      String uri = Strings.nullToEmpty(message.getAttribute(Attribute.URI));
+      String[] uriParts = uri.split("/");
+
+      if (keyStore == null) {
+        // If configured resources aren't available, this throws UncheckedIOException;
+        // this is unretryable so we allow it to bubble up and kill the worker and eventually fail
+        // the pipeline.
+        keyStore = KeyStore.of(metadataLocation.get(), kmsEnabled.get());
+      }
+
+      if (validator == null) {
+        validator = new JsonValidator();
+        try {
+          telemetryAetSchema = JSONSchemaStore.readSchema(Resources
+              .toByteArray(Resources.getResource("account-ecosystem/telemetry.schema.json")));
+          structuredAetSchema = JSONSchemaStore.readSchema(Resources
+              .toByteArray(Resources.getResource("account-ecosystem/structured.schema.json")));
+        } catch (IOException e) {
+          // We let this problem bubble up and kill the worker.
+          throw new UncheckedIOException("Unable to load AET JSON schemas", e);
+        }
+      }
+
+      ObjectNode json;
+      try {
+        json = Json.readObjectNode(message.getPayload());
+      } catch (IOException e) {
+        throw new UnparsableAetPayloadException(e);
+      }
+
+      // Note that we must work with the raw URI because ParseUri can raise errors that would
+      // route payloads containing anon_id values to error output; this transform must be placed
+      // before ParseUri so that we can handle redacting anon_id values.
```

Comment:
However, if the values are redacted at this point in the pipeline, it should be fine to pass the uri's into parse uri wholesale?